### ISSUE TITLE
Custom confirm template

### ISSUE THREAD

**Issue by Sunno** [caffeinehit/django-oauth2-provider#127]
Hi! I'd like to know if there's a way to set a custom template, since I've been reading docs but I don't find anything related.

**Comment by blag** [comment]
You should be able to override `provider/authorize.html` template with the vanilla Django template override mechanisms. IIRC, the rest of the views simply return redirects or HTTP error codes, so there isn't much point in customizing those templates.

Please note that this project is dead - use Django OAuth Toolkit if you can. I am on mobile so please forbive me for any spelling errors and for not linking to DOT

**Comment by Sunno** [comment]
I found a way, overriding url and specifing my own template.

This line must be placed before oauth urls

```python
url(r'^oauth2/authorize/confirm/?$', login_required(Authorize.as_view(template_name='oauth2/authorize.html'))),  # This one overrides template.
url(r'^oauth2/', include('provider.oauth2.urls', namespace='oauth2')),  # OAuth URLs
```

Actually I'm planning to switch to Oauth Toolkit since it has some options I need, I didn't know this project has dead. I'm using this since I found it in tastypie cookbook section.

**Comment by blag** [comment]
Yep, that is the vanilla Django template override mechanism. :+1:

Do you remember which section of the Tastypie, cookbook section that linked here? Sounds like their documentation could use an update.

Also, the Tastypie author [has expressed some concerns with the future of Tastypie](http://toastdriven.com/blog/2014/may/23/state-tastypie/), so you may want to consider using an alternative REST framework like [DRF](http://www.django-rest-framework.org/tutorial/quickstart/) or [Restless](https://github.com/toastdriven/restless).

**Comment by Sunno** [comment]
Oh, I didn't know it's called "vanilla Django template override mechanism", good to know :+1:

And about cookbok, they refer to an old tutorial [here](http://django-tastypie.readthedocs.org/en/latest/cookbook.html#creating-a-full-oauth-2-0-api)

Actually we're planning to migrate to django rest framework. But I'm still making some changes using old libraries meanwhile since project owner needs to show a demo soon.

**Comment by blag** [comment]
It's not really called that, it's just the normal way to override templates with Django. I'll see what I can do about updating that documentation to point elsewhere. The fewer links to this (dead) project, the better.

And if anybody has the proper access, this issue can be closed.
